webapp/src/ts/services/target-aggregates.service.ts

```diff
@@ -245,11 +250,12 @@ export class TargetAggregatesService {
     return places?.length ? places[0] : undefined;
   }
 
-  private getSupervisedContacts() {
+  private getSupervisedContacts(facilityId?) {
```

Comment:
So, the logic in this function is not properly handling my test setup.  I have the following contact hierarchy:

- `district_hospital` - My Supervisor's Health Facility
  - `person` - My Supervisor
  - `health_facility` - My CHW's Area
    - `person` - My CHW
    - `clinic` - My First Woman's Household
      - `person` - My First Womann
    - `clinic` - Primary Muted Person's Household
      - `person` - Primary Muted Person
  - `health_facility` - Second CHW's Area
    - `person` - Second CHW
    - `person` - Second Supervisor

I have two `chw` users associated with "My CHW" and "Second CHW" respectively.  Then I have two supervisor users.  One has a contact of "My Supervisor" and a facility of "My Supervisor's Health Facility".  The other supervisor has a contact of "Second Supervisor" and has two facilities: "My CHW's Area" and "Second CHW's Area".  

Things work fine for "My Supervisor" and the aggregate targets pull though as expected from both CHWs:
![image](https://github.com/user-attachments/assets/e1612a2b-024d-4bf2-9895-7099e29da12c)

For "Second Supervisor", though, I can see all the expected contacts from both CHW areas in the Person tab, and when I go to Targets, I can choose between the areas in the filter, but the display of the target aggregate data is all messed up and the list of supervised contacts is wrong:

![image](https://github.com/user-attachments/assets/cab44018-6bfb-48f3-90f9-191d0721012e)

After some debugging I can see that the display issued are caused by this function returning unexpected contact data.  Basically this function:

- takes the provided user facility (or gets the default for the user)
- Gets the direct _place children_ for that place
- Returns the primary contact's for those places as the "supervised contacts"

However the place children of "My CHW's Area" are just the households and so it is the primary contacts of those households that are getting pulled though as the "supervised contacts".

Reply:
@jkuester and I had a call and clarified this was the expected behavior when you have a supervisor at the CHW Area. Next steps are to update the docs to make it clear how Aggregate targets works in relation to the hierarchy level of a supervisor in the Doc site.

Reply:
Just to add more context for anyone reading in the future, in general the target aggregates functionality would not work for my "Second Supervisor" because the target aggregates only works for a specific hierarchy structure (where the supervisor contact is a sibling of the place contact that is the parent of the CHW).